charts/templates/statefulset.yaml

```diff
@@ -60,6 +61,50 @@ spec:
             {{- if .Values.extraContainerPorts }}
               {{- toYaml .Values.extraContainerPorts | nindent 12 }}
             {{- end }}
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                  - bash
+                  - '-c'
+                  - |
+                    TIMESTAMP=$(date +"%Y-%m-%d %T.%3N")
```

Comment:
The TIMESTAMP is never updated so all log messages will have the same time even though there are multiple calls and sleeps. That would be confusing to analyze. You could make a log function at the top:
```
log () {
  local TIMESTAMP=$(date +"%Y-%m-%d %T.%3N")
  echo "${TIMESTAMP} Info: $@" > /proc/$pid/fd/1
}
```

Reply:
Ya got it. Will update it